Sources_v3/StreamChatUI/Common Views/CurrentChatUserAvatarView.swift

```diff
@@ -0,0 +1,112 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+open class CurrentChatUserAvatarView<ExtraData: UIExtraDataTypes>: UIControl {
+    // MARK: - Properties
+    
+    public let uiConfig: UIConfig<ExtraData>
+    
+    public var controller: _CurrentChatUserController<ExtraData>? {
+        didSet {
+            controller?.setDelegate(self)
+            updateContent()
+        }
+    }
+    
+    // MARK: - Subviews
+    
+    public private(set) lazy var avatarView: AvatarView = {
+        let avatar = uiConfig.currentUser.currentUserAvatarView.init()
+        avatar.translatesAutoresizingMaskIntoConstraints = false
+        return avatar
+    }()
+    
+    // MARK: - Overrides
+    
+    open var defaultIntrinsicContentSize: CGSize?
+    override open var intrinsicContentSize: CGSize {
+        defaultIntrinsicContentSize ?? super.intrinsicContentSize
+    }
+    
+    override open var isHighlighted: Bool {
+        get { super.isHighlighted }
+        set { super.isHighlighted = newValue; updateContent() }
+    }
+    
+    override open var isSelected: Bool {
+        get { super.isSelected }
+        set { super.isSelected = newValue; updateContent() }
+    }
+    
+    override open var isEnabled: Bool {
+        get { super.isEnabled }
+        set { super.isEnabled = newValue; updateContent() }
+    }
```

Comment:
Can't we just use `addTarget(self, action: #selector(updateContent), for: .allEvents)` and update the content there?

Reply:
fixed in dff94e4bd93179075c8351a22fcc7bba2fa70e72

Reply:
Rolled back in af50f91a41cb2cb784f4d76a0f5802a520564c56: registering the selector (doesn't matter for **.allEvent** or **.touchEvents**) keeps view highlighted after `.touchUpInside` event